### DISCUSSION TITLE
[New Feature] Allow the AI to construct walls RA2-style

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1482] [Feature Requests]
### Description:
TS has an unused system for allowing the AI to build walls and gates around its base. It can be enabled by the rules.ini keys `AIBuildsWalls` (GDI) and `NodAIBuildsWalls` (Nod). However, the result is subpar, as the AI will often place disconnected wall sections in and around its base that offer little to no protection. It also doesn't work very well with the component tower logic.

RA2 and YR have a simpler but better system where the AI builds walls only around key structures like Construction Yards, superweapons, and Tech Centers.

### Possible Implementation:
Replace the unfinished/broken AI wall-building logic with RA2's.

**Comment by JoyfulShush** [comment]
This system should be customizable per building, rather than be hardcoded to specific building types.

**Reply by ZivDero** [reply]
I believe there's a tag in RA2 that makes the AI do it. The actual feature happening is kinda unlikely, though.

**Reply by JoyfulShush** [reply]
Why's that? Due to ground deformations and slopes?

**Reply by ZivDero** [reply]
Low priority combined with wall code being pretty obscure.
